src/common/admin_socket.cc

```diff
@@ -693,6 +694,140 @@ class GetdescsHook : public AdminSocketHook {
   }
 };
 
+// Define a macro to simplify adding signals to the map
+#define ADD_SIGNAL(signalName)                 \
+  {                                            \
+    ((const char*)#signalName) + 3, signalName \
+  }
+
+static const std::map<std::string, int> known_signals = {
+  ADD_SIGNAL(SIGHUP),
+  ADD_SIGNAL(SIGINT),
+  ADD_SIGNAL(SIGQUIT),
+  ADD_SIGNAL(SIGILL),
+  ADD_SIGNAL(SIGTRAP),
+  ADD_SIGNAL(SIGABRT),
+  ADD_SIGNAL(SIGBUS),
+  ADD_SIGNAL(SIGFPE),
+  ADD_SIGNAL(SIGKILL),
+  ADD_SIGNAL(SIGUSR1),
+  ADD_SIGNAL(SIGSEGV),
+  ADD_SIGNAL(SIGUSR2),
+  ADD_SIGNAL(SIGPIPE),
+  ADD_SIGNAL(SIGALRM),
+  ADD_SIGNAL(SIGTERM),
+  ADD_SIGNAL(SIGCHLD),
+  ADD_SIGNAL(SIGCONT),
+  ADD_SIGNAL(SIGSTOP),
+  ADD_SIGNAL(SIGTSTP),
+  ADD_SIGNAL(SIGTTIN),
+  ADD_SIGNAL(SIGTTOU)
+  // Add more signals as needed...
+};
+
+#undef ADD_SIGNAL
+
+class RaiseHook: public AdminSocketHook {
+  std::optional<pid_t> forked_child;
+public:
+  static const char* get_cmddesc()
+  {
+    return "raise "
+            "name=signal,type=CephString,req=true "
+            "-- "
+            "name=after,type=CephFloat,range=0.0,req=false";
+  }
+
+  static const char* get_help()
+  {
+    return "deliver the <signal> to the daemon process, optionally delaying <after> seconds; "
+            "when --after is used, the program will fork before sleeping, which allows to "
+            "schedule signal delivery to a suspened daemon; "
+            "<signal> can be in the forms '9', '-9', 'KILL', '-KILL'";
+  }
+
+  int call(std::string_view command, const cmdmap_t& cmdmap,
+      const bufferlist&,
+      Formatter* f,
+      std::ostream& errss,
+      bufferlist& out) override
+  {
+    string sigdesc;
+    ceph_assert(cmd_getval(cmdmap, "signal", sigdesc));
+    std::transform(sigdesc.begin(), sigdesc.end(), sigdesc.begin(),
+        [](unsigned char c) { return std::toupper(c); });
+    if (sigdesc.starts_with("-")) {
+      sigdesc.erase(0, 1);
+    }
+    if (sigdesc.starts_with("SIG")) {
+      sigdesc.erase(0, 3);
+    }
+    auto sig_it = known_signals.find(sigdesc);
+
+    int signal_to_send = 0;
+    pid_t recipient = getpid();
+
+    if (sig_it != known_signals.end()) {
+      signal_to_send = sig_it->second;
+      errss << "will send SIG" << sig_it->first << " (" << signal_to_send << ")" << std::endl;
+    } else if (signal_to_send = atoi(sigdesc.c_str())) {
+      errss << "will send signal " << signal_to_send << std::endl;
+    } else if (sigdesc == "L") {
+      f->open_object_section("known_signals");
+      for (auto& [name, num] : known_signals) {
+        f->dump_int(name, num);
+      }
+      f->close_section();
```

Comment:
This is useful but not documented (in the help or otherwise).

Reply:
It is documented in an error message when the signal isn't recognized. I'll consider adding to the docs.

Reply:
Documented also in the help